coalib/bearlib/abstractions/Linter.py

```diff
@@ -316,18 +324,29 @@ def _convert_output_regex_match_to_result(self,
                 groups['origin'] = '{} ({})'.format(klass.__name__,
                                                     groups['origin'].strip())
 
-            # Construct the result.
-            return Result.from_values(
-                origin=groups.get('origin', self),
-                message=(groups.get('message', '').strip()
-                         if result_message is None else result_message),
-                file=filename,
-                severity=groups['severity'],
-                line=groups['line'],
-                column=groups['column'],
-                end_line=groups['end_line'],
-                end_column=groups['end_column'],
-                additional_info=groups.get('additional_info', '').strip())
+            # GlobalBears do not pass a filename to the function. But they can
+            # still give one through the regex
+            if filename is None:
+                filename = groups.get('filename', None)
```

Comment:
to make this working properly you have to add `filename` to the list of supported names if global-bear shall be used: https://github.com/coala/coala/pull/3652/commits/495db77764bb0bd0344599e867bd44957f5b511e#diff-6dee3b7ded9b28fd0db83b2e7d039b86R66